### ISSUE TITLE
Support for RMSProp as a Learning Rule

### ISSUE THREAD

**Issue by madisonmay** [lisa-lab/pylearn2#1023]
Figured I'd submit a ticket for support for the adaptive learning rate method RMSProp as described by Geoffrey Hinton in his U. Toronto Neural Networks class.  

In short, RMSProp involves keeping a moving average of the squared gradient for each weight, and dividing the gradient for each weight by square root of the moving average at each time step.  RMSProp is a variant of RProp designed to play nicely with minibatches.   

Implementation details available in the second half of the slide deck below:
http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf
